application/cms/views.py

```diff
@@ -1176,6 +1190,71 @@ def create_data_source(topic_slug, subtopic_slug, measure_slug, version):
         )
 
 
+@cms_blueprint.route("/<topic_slug>/<subtopic_slug>/<measure_slug>/<version>/edit/data-sources/link", methods=["POST"])
+@login_required
+@user_has_access
+def link_existing_data_source(topic_slug, subtopic_slug, measure_slug, version):
+    topic, subtopic, measure, measure_version = page_service.get_measure_version_hierarchy(
+        topic_slug, subtopic_slug, measure_slug, version
+    )
+
+    form = SelectOrCreateDataSourceForm(data_sources=[])
+
+    if len(measure_version.data_sources) >= 2:
```

Comment:
Hmm, feels a bit icky to have this logic in the view (rather than a model or a form validator), but I can't think of an easy alternative right now...